staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -249,31 +219,29 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 		return nil
 	})
 
-	s.GenericAPIServer.PrepareOpenAPIService()
-
-	if s.GenericAPIServer.OpenAPIService != nil {
-		s.rootSpec = s.GenericAPIServer.OpenAPIService.GetSpec()
-		if err := s.updateOpenAPISpec(); err != nil {
+	if openApiConfig != nil {
+		s.openAPIAggregator, err = buildAndRegisterOpenAPIAggregator(
+			s.delegateHandler,
```

Comment:
What about the delegate of the delegate? The chain is not necesserily depth 1. Already today we have the apiextensions server behind the main kube-apiserver.